x-pack/plugins/security_solution/public/common/components/event_details/overview/index.tsx

```diff
@@ -0,0 +1,178 @@
+
+    const riskScoreData = useMemo(() => {
+      const item = find({ field: 'kibana.alert.rule.risk_score', category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    const ruleNameData = useMemo(() => {
+      const item = find({ field: SIGNAL_RULE_NAME_FIELD_NAME, category: 'kibana' }, data);
+      const linkValueField = find({ field: 'kibana.alert.rule.uuid', category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+          linkValueField,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    return (
+      <NotGrowingFlexGroup gutterSize="s">
+        {hasData(statusData) && (
+          <EuiFlexItem>
+            <OverviewCard title={SIGNAL_STATUS}>
+              <StatusPopoverButton
+                eventId={eventId}
+                contextId={contextId}
+                enrichedFieldInfo={statusData}
+                indexName={indexName}
+                timelineId={timelineId}
+                handleOnEventClosed={handleOnEventClosed}
+              />
+            </OverviewCard>
+          </EuiFlexItem>
+        )}
+
+        {hasData(severityData) && (
+          <EuiFlexItem>
+            <OverviewCardWithActions
+              title={ALERTS_HEADERS_SEVERITY}
+              enrichedFieldInfo={severityData}
+              contextId={contextId}
+            >
+              <SeverityBadge value={severityData.values[0]} />
+            </OverviewCardWithActions>
+          </EuiFlexItem>
+        )}
+
+        {hasData(riskScoreData) && (
+          <EuiFlexItem>
+            <OverviewCardWithActions
+              title={ALERTS_HEADERS_RISK_SCORE}
+              enrichedFieldInfo={riskScoreData}
+              contextId={contextId}
+            >
+              {riskScoreData.values[0]}
+            </OverviewCardWithActions>
+          </EuiFlexItem>
+        )}
+
+        {hasData(ruleNameData) && (
+          <EuiFlexItem>
+            <OverviewCardWithActions
+              title={ALERTS_HEADERS_RULE}
+              enrichedFieldInfo={ruleNameData}
+              contextId={contextId}
+              alignItems="flexEnd"
+            >
+              <FormattedFieldValue
+                contextId={contextId}
+                eventId={eventId}
+                value={ruleNameData.values[0]}
+                fieldName={ruleNameData.data.field}
+                linkValue={ruleNameData.linkValue}
+                fieldType={ruleNameData.data.type}
+                fieldFormat={ruleNameData.data.format}
+                isDraggable={false}
+                truncate={false}
```

Comment:
consider applying truncation here to avoid [subtle issues with very long rule names](https://github.com/elastic/kibana/issues/119012)

Reply:
I tried to use that but it had some strange side-effects where it prevented the text from wrapping:
![Screenshot 2021-12-07 at 14 29 17](https://user-images.githubusercontent.com/68591/145038128-28b6c9b0-1b4c-4ce8-8e81-16dd357148b5.png)

That was my motivation to introduce this helper component that limits the text to two lines: https://github.com/elastic/kibana/pull/120347/commits/c2a1bf633a7f410953e7d71a6cb13c1b27cb9c98